api/v1alpha1/modelregistry_webhook.go

```diff
@@ -159,6 +175,9 @@ func (r *ModelRegistry) Default() {
 
 // ValidateRegistry validates registry spec
 func (r *ModelRegistry) ValidateRegistry() (warnings admission.Warnings, err error) {
+	// set runtime defaults before validation, just like the reconcile loop
+	r.RuntimeDefaults()
```

Comment:
Overall looks good to me, my only doubt is this line.
Can you help me understand why we are mutating inside the validation webhook?

Reply:
To make sure that the resource will be valid after applying runtime defaults. In case you are wondering about it, the changes in validating webhook are actually not propagated back to k8s.

Reply:
Basically, runtime defaults are in memory and not saved in the CR.

Reply:
thanks, confirming my /lgtm